mso/resource_mso_schema_site_contract_service_graph_listener.go

```diff
@@ -0,0 +1,918 @@
+							}, false),
+						},
+						"redirect_port": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+						"redirect_code": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"unknown",
+								"permMoved",
+								"found",
+								"seeOther",
+								"temporary",
+							}, false),
+						},
+						"health_check": &schema.Schema{ // Only one object allowed
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"port": &schema.Schema{
+										Type:     schema.TypeInt,
+										Optional: true,
+										Computed: true,
+									},
+									"protocol": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+										Computed: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											"http",
+											"https",
+											"tcp",
+											"udp",
+											"tls",
+											"inherit",
+										}, false),
+									},
+									"path": &schema.Schema{
+										Type:         schema.TypeString,
+										Optional:     true,
+										Computed:     true,
+										ValidateFunc: validation.StringLenBetween(1, 1000),
+									},
+									"interval": &schema.Schema{
+										Type:     schema.TypeInt,
+										Optional: true,
+										Computed: true,
+									},
+									"timeout": &schema.Schema{
+										Type:     schema.TypeInt,
+										Optional: true,
+										Computed: true,
+									},
+									"unhealthy_threshold": &schema.Schema{
+										Type:     schema.TypeInt,
+										Optional: true,
+										Computed: true,
+									},
+									"use_host_from_rule": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+										Computed: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											"yes",
+											"no",
+										}, false),
+									},
+									"success_code": &schema.Schema{
+										Type:         schema.TypeString,
+										Optional:     true,
+										Computed:     true,
+										ValidateFunc: validation.StringLenBetween(1, 1000),
+									},
+									"host": &schema.Schema{
+										Type:         schema.TypeString,
+										Optional:     true,
+										Computed:     true,
+										ValidateFunc: validation.StringLenBetween(1, 1000),
+									},
+								},
+							},
+						},
+						"target_ip_type": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"unspecified",
+								"primary",
+								"secondary",
+							}, false),
+						},
+					},
+				},
+			},
+		},
+		CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
```

Comment:
Can you comment why the customizeDiff is required? Think that would make it easier for people to understand.